applications/StructuralMechanicsApplication/custom_response_functions/response_utilities/adjoint_postprocess.cpp

```diff
@@ -0,0 +1,609 @@
+            {
+                auto it = mrModelPart.ElementsBegin() + i;
+                it->SetValue(variable_pair[1], variable_pair[1].Zero());
+            }
+        }
+        // Set conditional sensitivity result variables to zero.
+        for (const auto& variable_pair : mConditionSensitivityScalarVariables)
+        {
+            #pragma omp parallel for
+            for (int i = 0; i< static_cast<int> (mrModelPart.NumberOfConditions()); ++i)
+            {
+                auto it = mrModelPart.ConditionsBegin() + i;
+                const SizeType number_of_nodes = it->GetGeometry().size();
+                for(IndexType j = 0; j < number_of_nodes; ++j)
+                    it->GetGeometry()[j].FastGetSolutionStepValue(variable_pair[1]) = variable_pair[1].Zero();
+            }
+        }
+        for (const auto& variable_pair : mConditionSensitivityVectorVariables)
+        {
+            #pragma omp parallel for
+            for (int i = 0; i< static_cast<int> (mrModelPart.NumberOfConditions()); ++i)
+            {
+                auto it = mrModelPart.ConditionsBegin() + i;
+                const SizeType number_of_nodes = it->GetGeometry().size();
+                for(IndexType j = 0; j < number_of_nodes; ++j)
+                    it->GetGeometry()[j].FastGetSolutionStepValue(variable_pair[1]) = variable_pair[1].Zero();
+            }
+        }
+
+        KRATOS_CATCH("");
+    }
+
+    void AdjointPostprocess::UpdateSensitivities()
+    {
+        KRATOS_TRY;
+
+        if (mBuildMode == "static")
+        {
+            // overwrite existing.
+            this->SetAllSensitivityVariablesToZero();
+        }
+        else
+        {
+            KRATOS_ERROR << "Unsupported \"build_mode\": " << mBuildMode << std::endl;
+        }
+
+        for (const auto& variable_pair : mNodalSensitivityScalarVariables)
+            this->UpdateNodalSensitivities(variable_pair[0], variable_pair[1]);
+        for (const auto& variable_pair : mNodalSensitivityVectorVariables)
+            this->UpdateNodalSensitivities(variable_pair[0], variable_pair[1]);
+        for (const auto& variable_pair : mElementSensitivityScalarVariables)
+            this->UpdateElementSensitivities(variable_pair[0], variable_pair[1]);
+        for (const auto& variable_pair : mElementSensitivityVectorVariables)
+            this->UpdateElementSensitivities(variable_pair[0], variable_pair[1]);
+        for (const auto& variable_pair : mConditionSensitivityScalarVariables)
+            this->UpdateConditionSensitivities(variable_pair[0], variable_pair[1]);
+        for (const auto& variable_pair : mConditionSensitivityVectorVariables)
+            this->UpdateConditionSensitivities(variable_pair[0], variable_pair[1]);
+
+        KRATOS_CATCH("");
+    }
+
+    template <typename TDataType>
+    void AdjointPostprocess::UpdateNodalSensitivities(Variable<TDataType> const& rSensitivityVariable, Variable<TDataType> const& rOutputVariable)
```

Comment:
I will refactor it.

Reply:
For similar reason as above, I would suggest we don't spend a lot of time making this perfect since we need replace a lot of it with the core implementation.